### PR TITLE
ta: pkcs11: use TEE UUID and toekn ID in reported token description

### PR DESCRIPTION
**Pull request by etienne-lms**
Changes suggested and implemented by @ricardosalveti, and ported back in the PKCS11 to use OP-TEE UUID in token description and token index in token default serial number.

### CODE REVIEW

**Comment by ricardosalveti** [comment]
+1 from my side, thanks so much for getting the changes!

File: ta/pkcs11/src/pkcs11_token.c [hunk 1]
```diff
@@ -309,6 +309,12 @@ uint32_t entry_ck_token_info(uint32_t ptypes, TEE_Param *params)
 
 	pad_str(info.manufacturer_id, sizeof(info.manufacturer_id));
 	pad_str(info.model, sizeof(info.model));
+
+	if (snprintf(sn, sizeof(sn), "%0*"PRIu32,
+		     PKCS11_TOKEN_SERIALNUM_SIZE, token_id) >= (int)sizeof(sn))
+		TEE_Panic(0);
```
Comment (etienne-lms) [inline ta/pkcs11/src/pkcs11_token.c:315]: maybe should more strictly test `snprintf()` does not return a negative value...

File: ta/pkcs11/src/pkcs11_token.c [hunk 2]
```diff
@@ -189,6 +189,31 @@ static void pad_str(uint8_t *str, size_t size)
...
+{
+	char desc[sizeof(info->slot_description) + 1] = { 0 };
+	TEE_UUID dev_id = { };
+	TEE_Result res = TEE_ERROR_GENERIC;
+	int n = 0;
+
+	res = TEE_GetPropertyAsUUID(TEE_PROPSET_TEE_IMPLEMENTATION,
+				    "gpd.tee.deviceID", &dev_id);
+	if (res == TEE_SUCCESS) {
+		n = snprintf(desc, sizeof(desc), PKCS11_SLOT_DESCRIPTION
```
Comment (jenswi-linaro) [inline ta/pkcs11/src/pkcs11_token.c:203]: `snprintk()` is needed for `%pUl` to print a string representation of a UUID.

File: ta/pkcs11/src/pkcs11_token.c (hunk 1 above)
Comment (jenswi-linaro) [reply ta/pkcs11/src/pkcs11_token.c:315]: +1

File: ta/pkcs11/src/pkcs11_token.c (hunk 2 above)
Comment (etienne-lms) [reply ta/pkcs11/src/pkcs11_token.c:203]: ok. 
That's for consistency I guess? in optee both `snprintk` and `snprintf` end in the same `__vsnprintf`.

Comment (jenswi-linaro) [reply ta/pkcs11/src/pkcs11_token.c:203]: Yes, but with a small difference, with `ext` passed as `true` or `false`.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by Emantor** [comment]
`Acked-by: Rouven Czerwinski <r.czerwinski@pengutronix.de>`

**Comment by etienne-lms** [comment]
fixup commits squashed.
tags applied.

**Comment by etienne-lms** [comment]
fixup commit squashed.
commit log message fixed.
tag applied.
thanks

**Comment by jforissier** [comment]
It looks like a `(int)` cast is needed.

**Comment by jenswi-linaro** [comment]
Or rather the correct format specifier, `%zu`.

**Comment by jforissier** [comment]
Etienne said he's having troubles with his laptop and I could apply the needed fix and merge manually -- which is what I just did.

**Comment by etienne-lms** [comment]
many thanks @jforissier 👍
